benchmark_runner/jupyterlab/templates/summary_report/summary_report_operations.py

```diff
@@ -0,0 +1,353 @@
+                               'geometric_mean': []}
+        elif workload == 'uperf':
+            # Filter out values greater than 1000
+            df = df[(df['norm_ltcy'] < 1000) & df['norm_ltcy'].notna()]
+            self.geometric_mean_data = {'uuid': [], 'ocp_version': [], 'cnv_version': [], 'odf_version': []}
+            # Separate object to latency and throughput for display it separately
+            self.geometric_mean_data_latency = {'workload': workload, 'uuid': [], 'metric': 'latency', 'ocp_version': [], 'cnv_version': [],
+                                           'odf_version': [], 'geometric_mean': []}
+            self.geometric_mean_data_throughput = {'workload': workload, 'uuid': [], 'metric': 'throughput', 'ocp_version': [], 'cnv_version': [],
+                                              'odf_version': [], 'geometric_mean': []}
+        elif workload == 'vdbench':
+            df = df[(df['Run'] != 'fillup')]
+            self.geometric_mean_data = {'uuid': [], 'ocp_version': [], 'cnv_version': [], 'odf_version': []}
+            # Separate object to latency and throughput for display it separately
+            self.geometric_mean_data_iops = {'workload': workload, 'uuid': [], 'metric': 'iops', 'ocp_version': [], 'cnv_version': [],
+                                        'odf_version': [], 'geometric_mean': []}
+            self.geometric_mean_data_latency = {'workload': workload, 'uuid': [], 'metric': 'latency', 'ocp_version': [], 'cnv_version': [],
+                                           'odf_version': [], 'geometric_mean': []}
+        elif workload == 'bootstorm' or workload == 'windows':
+            self.geometric_mean_data = {'workload': 'bootstorm', 'uuid': [], 'metric': [], 'ocp_version': [],'cnv_version': [], 'odf_version': [], 'geometric_mean': []}
+
+        return df
+
+    @staticmethod
+    def __calc_percentage(df: pd.DataFrame, complementary: bool = False):
+        """
+        This method calculates percentage difference
+        @param df:
+        @param complementary: calculates the complementary percentage change, meaning it measures the difference from 100%
+        @return:
+        """
+        # complementary percentage change: e.g. latency
+        if complementary:
+            df['result'] = df.groupby('metric')['geometric_mean'].transform(lambda x: 1 - ((x - x.shift(1)) / x.shift(1)) * 100)
+        else:
+            df['result'] = df.groupby('metric')['geometric_mean'].transform(lambda x: ((x - x.shift(1)) / x.shift(1)) * 100)
+        df = df.dropna(subset=['result'])
+        df.loc[:, 'result'] = df['result'].round(2)
+        return df[['workload', 'metric', 'result']]
+
+    @typechecked()
+    def aggregate_workload_dataframe(self, workload: str):
+        """
+        This method aggregates workload dataframe and updates metric values
+        @param workload:
+        @return:
+        """
+        if workload == 'hammerdb':
```

Comment:
Delegate this to separate functions and use a dict of workload name and aggregating function.  This function is too long and difficult to maintain.

Reply:
I dont see any reason to move it to separate methods, we just prepare the workload dataframe template and its different between each workload.

Reply:
Right, but if you want to add more workloads it means adding more inline code here.  I won't reject it for this, but...

Reply:
Fixed it